src/controllers/user.controller.ts

```diff
@@ -80,11 +86,58 @@ export const deleteUser = async (req: Request, res: Response) => {
     if (!user) {
       return res.status(404).json({ message: 'User not found' });
     }
-    await user.destroy();
+    await removeProfile(id)
+    await user.destroy();   
     res.json({ message: 'User deleted successfully' });
   } catch (error: unknown) {
     if (error instanceof Error) {
       res.status(500).json({ message: error.message });
     }
   }
 };
+
+// profile part 
+
+export const getProfile = async (req: Request, res: Response): Promise<void> => {
+  try {
+    const { id } = req.params;
+    const profile = await Profile.findOne({
+      where: {
+        userId: id,
+      },
+    });
```

Comment:
There is a service responsible get or query a single user called querySingleUser. use that @shebz2023